### DISCUSSION TITLE
Migration to MaterialTimePicker

### DISCUSSION THREAD

**Discussion by hiqua** [iSoron/uhabits#865] [Technical Discussion]
In SnoozeDelayPickerActivity.kt, changing `showTimerPicker` to this:
```kt
    private fun showTimePicker() {
        val timeFormat = if (DateFormat.is24HourFormat(this)) {
            TimeFormat.CLOCK_24H
        } else {
            TimeFormat.CLOCK_12H
        }
        val calendar = Calendar.getInstance()
        val picker = MaterialTimePicker.Builder()
                .setTitleText(getString(R.string.next_reminder))
                .setTimeFormat(timeFormat)
                .setHour(calendar[Calendar.HOUR_OF_DAY])
                .setMinute(calendar[Calendar.MINUTE])
                .build()
        picker.addOnPositiveButtonClickListener {
            reminderController!!.onSnoozeTimePicked(habit, picker.hour, picker.minute)
            finish()
        }

        picker.show(supportFragmentManager, "timePicker")
    }
```

yields similar results to the current state, without using the vendored library.
It still requires customization for theming (it also doesn't work out of the box without removing the `android:theme`). That would be a step towards getting rid of the vendored library.

Would that be a good way forward?

**Comment by iSoron** [comment]
@hiqua Sure, feel free to submit a PR! It would be nice to completely remove all vendored code. The only reason we copied the code directly from AOSP is because the code was not available as a library at the time.
